lib/spack/spack/schema/packages.py

```diff
@@ -59,6 +59,15 @@
                             'type':  'boolean',
                             'default': True,
                         },
+                        'readable': {
+                            'type':  'string',
```

Comment:
These should only allow the legal values of `user`, `group`, and `world`, then you don't have to do all that error checking and `ConfigFileError`s will be raised automatically.